### ISSUE TITLE
Suspended processes on Windows reported as running

### ISSUE THREAD

**Issue by mfujitaitrs** [oshi/oshi#1581]
As seen on Windows task manager this particular process' status is Suspended while the reported status from Oshi is Running.
![image](https://user-images.githubusercontent.com/42991925/112279869-5a815680-8cbf-11eb-8681-75e07778d811.png)

Processed value from Oshi:
`process_id=1048, process_name=IntelCpHDCPSvc, Name=processes_state, Value=RUNNING@STRING`

**Comment by dbwiddis** [comment]
Per the comments in the source code:
```
        // There is no easy way to get ExecutuionState for a process.
        // The WMI value is null. It's possible to get thread Execution
        // State and possibly roll up.
        this.state = RUNNING;
```
PR welcome if this value is important.

**Comment by dbwiddis** [comment]
When this was originally written we didn't have thread details in the API.  It's probably not too hard to make a configurable (defaulting "on") option to call `getThreadDetails()` and iterate over the thread states and return the appropriate state (e.g., if any thread is running it's running).

PR welcome.

**Comment by dbwiddis** [comment]
Digging around, It seems to be a pretty simple rule. Task manager only has 3 possible process states: Unknown, Running, and Suspended.  Windows processes are considered "running" unless every one of their threads has a WaitReason of Suspended.  (Actual code: set suspended; if any thread is *not* suspended set running and exit.)  See [this structure](https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ex/sysinfo/thread.htm) relevant to this calculation.

[This enum](http://www.nirsoft.net/kernel_struct/vista/KWAIT_REASON.html) has the value to test.

Unfortunately to get that value it's either internal code or as a result of the "we might remove this" [NtQuerySystemInformation](https://docs.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntquerysysteminformation) function.

BUT!  We are in luck, the data we retrieve for thread details for Windows comes from the performance counters, which we fetch from the registry or as a backup, the [Win32_PerfRawData_PerfProc_Thread](https://docs.microsoft.com/en-us/previous-versions/aa394325(v=vs.85)) WMI class, and they expose the `ThreadWaitReason`, noting it's only applicable if the thread is in the "Wait" state.

So it is possible to do, but it's a lot of data to grep through just for the very rare case of a suspended process, and probably too big a performance hit in the normal case.  But the data is available for the adventurous.

**Comment by dbwiddis** [comment]
Possible implementation I think we should consider:
 - Config file setting (defaulting "off") to check thread state
 - If config parameter set on:
   - Add a new value SUSPENDED to the Process/Thread State Enum 
   - Create similar methods to `ThreadPerformanceData.buildThreadMapFromRegistry()` and  `buildThreadMapFromPerfCounters()`, except:
     - We only need the process ID, thread state, and thread wait reason fields
     - We don't need to create a new OSThread object, what we want to return is a map of Process ID to State
     - Iterate all the threads
       - If the thread state is "Waiting" check the wait reason.  If "5" and the process ID is not yet in the map, put it in the map with state SUSPENDED.
       - Otherwise, if the process ID is either not in the map, or in the map with SUSPENDED state (if getordefault(pid, suspended)), change the state to RUNNING.
       - (We should also alter the existing threads to return a SUSPENDED state in the case of WAITING and reason 5.)
     - Return the map to the user
   - Use the map created above when instantiating an OSProcess to set the state appropriately.

**Comment by dbwiddis** [comment]
I've implemented this in #1592.

We already queried the information to get suspended Thread state, so that's in by default.  Checking all threads just to update processes is a significant performance hit so that's off by default, with a configuration option to turn it on.  You can still manually get thread details to see if they're all suspended.
